### ISSUE TITLE
Color picker in curves gives unexpected behavior when tool is disabled

### ISSUE THREAD

**Issue by Thanatomanic** [Beep6581/RawTherapee#4925]
The color picker to add/modify points to the curve in the Lab color regions tool does not function ~~at all~~ as expected (added:) when the tool is disabled. ~~Maybe this functionality hasn't been implemented, but until then disabling this tool would be necessary I think.~~ @agriggio any thoughts?

(image removed)

**Comment by Thanatomanic** [comment]
Oh, wait a minute. It does work properly, however only when the tool is enabled.

Still there is a bug hidden here, because even if the tool is disabled you can modify the first points in the curve when using Ctrl+click.

**Comment by agriggio** [comment]
there's a hack in the code to make it work, which is applied only if the tool is enabled, in order to avoid doing unnecessary work. if however this is expected to work also when the tool is disabled, I can do that. there will be a modest performance impact, but I think it won't be visible (essentialy you need to go through the `L*a*b*` tool even if disabled, because that's where the color picker data comes from

**Comment by heckflosse** [comment]
In other tools, Lab for example, pipette also don't work if the tool is disabled. I think it's ok the way it is now or we have to change the behaviour for other tools as well, which we should not do for 5.5 imho

**Comment by Thanatomanic** [comment]
I agree we should not do anything for 5.5. But I would think that the following is undesired behavior for curves: https://filebin.net/s2gan80l6984kwyc (sorry FLV only).
Note: you first need to click in the image before the first point in the curve becomes active.

**Comment by Hombre57** [comment]
The other way round, and possibly more logical, would be to enable the tool if you start editing a curve, because there's no point doing that when the tool is disabled. Probably we should make the tool unsensitive if disabled, and that would be easy to do (I guess).

**Comment by Hombre57** [comment]
@agriggio It seem you did the hack you're talking about and already committed ? On my side, I'm trying to make the disabled tools unsensitive, which should solve the issue too. I'll create a new branch when done.

On a side note, I'd like to know if I'm not too pedantic at code reorganization : I'm moving the functions definition (as opposed to their prototype) from `toolpanel.h` to `toolpanel.cc`, but should i do that for this kind of empty functions too : `virtual void setExpanded (bool expanded) {}` ?

**Comment by agriggio** [comment]
@Hombre57 yes, but the hack was to make the color pickers work with the new tool, not to handle the "tool disabled" case

**Comment by Hombre57** [comment]
@agriggio I saw this comment in your commits :

> // hack to fill the pipette buffers also when
> // !params->labCurve.enabled. It is ugly, but it's the smallest code
> // change that I could find

...but didn't saw the whole picture though. Have you remove the hack ?

**Comment by agriggio** [comment]
> Have you remove the hack ?

No, it's not possible (meaning: I don't know how to do it :-) to remove it without a significant refactoring, which I'm not willing to do at the moment...

**Comment by Hombre57** [comment]
Branch `tool-sensitivity` created. Tell me if you're fine with this behavior. If so, I'll add disconnection of the pipette and on preview editing.

**Comment by Floessie** [comment]
@Hombre57 
> On a side note, I'd like to know if I'm not too pedantic at code reorganization : I'm moving the functions definition (as opposed to their prototype) from toolpanel.h to toolpanel.cc, but should i do that for this kind of empty functions too : virtual void setExpanded (bool expanded) {} ?

I'm strict with my code in this respect, but your mileage may vary. Thing is: One day someone wants to experiment with `setExpanded()` and change the default implementation. Changing the header will lead to massive recompilation. So the stub has to be moved to the implementation anyway, and doing that from the beginning is a helping hand to one who has to deal with the code after you.
